packages/cells/src/widget.ts

```diff
@@ -1040,8 +1057,17 @@ export class CodeCell extends Cell<ICodeCellModel> {
       this.addClass(DIRTY_CLASS);
     }
 
-    this._outputPlaceholder = new OutputPlaceholder(() => {
-      this.outputHidden = !this.outputHidden;
+    this._outputPlaceholder = new OutputPlaceholder({
+      callback: () => {
+        this.outputHidden = !this.outputHidden;
+      },
+      text: this.getOutputPlaceholderText()
+    });
+
+    this.model.outputs.changed.connect((sender, args) => {
+      if (this._outputPlaceholder && this.outputHidden) {
+        this._outputPlaceholder.text = this.getOutputPlaceholderText();
+      }
```

Comment:
Could you add this in `onOutputChanged` (~line 1370) instead of hooking a lambda here?